abrams_strogatz/lang_comp_bilinguals.py

```diff
@@ -0,0 +1,305 @@
+        by a value -1) or laguage A and B (represented by a value 0).
+        The latter represent bilingual speakers.
+
+    """
+    popu = np.random.choice([1, 0, -1], size=(m, n), p=[pa, 1.0 - pa - pb, pb])
+    return popu
+
+
+def repres(popu: np.array) -> plt.figure:
+    """Graphical 2D-representation of the (mxn) array.
+
+    Each site represents an individual speaking either language A,
+    language B, or language A and B (bilinguals). Languages are pictured
+    by a selection of colors (blue, white, red).
+
+    Args:
+        popu: Array containing the language spoken by the population.
+            The values contained inside the array are 1 (lang A),
+            0 (bilingual) and -1 (lang B).
+
+    Returns:
+        Returns a mpl.figure object containing the graphical representation
+            of the array.
+
+    """
+    col = mpl.colors.ListedColormap(["Blue", "White", "Red"])
+    colbar_tick = np.array([-1, 0, 1])
+    fig = plt.figure()
+    ax = plt.axes()
+    plot = ax.matshow(popu, cmap=col, origin="lower", animated=True)
+    ax.xaxis.tick_bottom()
+    ax.xaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.yaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.xaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.yaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    fig.colorbar(plot, ax=ax, ticks=colbar_tick, label="Language").ax.set_yticklabels(
+        ["B", "AB", "A"],
+    )
+    # To avoid an excessive computation cost, the graphical
+    # representation of the lattice is not displayed. Only the figure instance
+    # is returned.
+    plt.close()
+    return fig
+
+
+def periodic_boundary(index: tuple, lattice_shape: tuple) -> tuple:
+    """Apply periodic boundary conditions.
+
+    We consider a regular lattice with periodic boundary conditions.
+    periodic_boundary function is used to apply this condition to
+    the selected node. If the node is located outside the boundaries,
+    the neighbors are selected applying usual nearest neighbor
+    conditions. If the node is located at the boundary, the function
+    will return a tuple where the neighbors are selected following
+    periodic boundary conditions (in this way, the regular lattice
+    becomes a torus).
+
+    Args:
+        index: 2D-tuple. Represents the selected node.
+        lattice_shape: 2D-tupple. Shape of the lattice.
+    Returns:
+        The functions returns a tuple where periodic boundary conditions
+        are applied (if needed).
+
+    Examples:
+    If the lattice is a 5x5 array, a node located at position (6, 5) will
+    result:
+
+    >>> print(periodic_boundary((6, 5), (5, 5)))
+    (1,0)
+    """
+    return tuple((i % s for i, s in zip(index, lattice_shape)))
+
+
+def language_dynamics(popu: np.array, m: int, n: int, s: float, a: float = 1.0) -> np.array:
+    """Population evolution. Change the language spoken by individuals.
```

Comment:
105:   1 D401 First line should be in imperative mood [flake8-docstrings]
```suggestion
    """
    Evolve the population of speakers following the Minett-Wang bilingual model.
```

Reply:
Checked and corrected all function docstrings. Now, all first lines are in imperative mood.